### PR TITLE
Modularize eslint config

### PR DESCRIPTION
**Pull request by jakemccloskey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+```json
+{
+    "extends": [
+        "1stdibs"
+    ],
+    "rules": {
+        "no-bitwise": 0
+    }
+}
```
Comment (hellatan) [inline README.md]: oh what if you want to add more rulesets to this repo? or is that all outlined in the eslint documentation?

Comment (jakemccloskey) [inline README.md]: It's all in the linked documentation.

**Comment by hellatan** [comment]
cc @twhid @robrichard @antialias @nameofname @jrdrg @prestifidibs @brettjashford @gedas-monginas @nitruxa @Tyriont 

I'm gonna merge this soon

**Comment by twhid** [comment]
:-1: do an npm install before linting :cry: 

The answer to this problem is scaffolding, not npm packages.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "eslint-config-1stdibs",
+  "github": "https://github.com/1stdibs/backbone-base-and-form-view/",
```
Comment (hellatan) [inline package.json]: this url is wrong

**Comment by prestifidibs** [comment]
You already have to npm install -g eslint before you can lint anyhow.  This way we could maintain the version of eslint everyone is using.

**Comment by antialias** [comment]
@twhid would scaffolding help us when we want to make a global change across all the configurations for bunsen, v2, and .com?

**Comment by twhid** [comment]
I repeat, scaffolding is the answer.

**Comment by twhid** [comment]
don't see why not..

**Comment by antialias** [comment]
[yeoman](http://yeoman.io/)?

**Comment by antialias** [comment]
As I understand it, scaffolding gives you a good head start, but once the project has started, you can't pull in updates from shared resources.

**Comment by prestifidibs** [comment]
I don't get it.  How is scaffolding the answer?

**Comment by twhid** [comment]
Sets you up with the lint config you want. As TH pointed out, we just need
a way to update configs with the scaffolding.

Just seems kinda crazy to have to do an entire npm install to lint your
code. Your point re: globally installing eslint could be made against every
single piece of software on your computer that isn't installed locally via
npm install

**Comment by antialias** [comment]
> Just seems kinda crazy to have to do an entire npm install to lint your
code

I experimented with having a separate set of deps that could be installed just for flightplan that addresses this concern. I still think it's not a bad way to handle the pre-build: https://github.com/1stdibs/1stdibs.com/pull/3384

**Comment by twhid** [comment]
Another option would be to track packages that we need for flightplan
and/or other things we'd like to run prior to a full install.

**Comment by prestifidibs** [comment]
Our pre-commit hook only works after an npm install + gulp.  Would this be solved with scaffolds, too?

**Comment by prestifidibs** [comment]
> Sets you up with the lint config you want.

How is this different than npm install?

**Comment by antialias** [comment]
> Another option would be to track packages that we need for flightplan
and/or other things we'd like to run prior to a full install.

What's exactly what the branch that I linked to in my prior comment does.

**Comment by antialias** [comment]
right here: https://github.com/1stdibs/1stdibs.com/pull/3384/files#diff-b8f3386edff9102f9712eaf309e8a2e5R17

**Comment by hellatan** [comment]
i understand that the debate is lint before install, but I think we should at the very minimum merge these settings in b/c then it can become our definitive source for linting. Walmart has a bunch of configs like this as well - https://github.com/walmartlabs/eslint-config-defaults

**Comment by hellatan** [comment]
merging these configs

**Pull request merged**